src/EditorFeatures/Core/Implementation/TodoComment/AbstractTodoCommentIncrementalAnalyzer.cs

```diff
@@ -91,9 +87,31 @@ public Task DocumentResetAsync(Document document, CancellationToken cancellation
             }
         }
 
+        private async Task<IList<TodoComment>> GetTodoCommentsAsync(Document document, ImmutableArray<TodoCommentDescriptor> tokens, CancellationToken cancellationToken)
+        {
+            var client = await _workspace.GetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
+            if (client != null && !document.IsOpen())
+            {
+                // run todo scanner on remote host
+                return await client.RunCodeAnalysisServiceOnRemoteHostAsync<IList<TodoComment>>(
+                    document.Project.Solution, nameof(IRemoteTodoCommentService.GetTodoCommentsAsync),
+                    new object[] { document.Id, tokens }, cancellationToken).ConfigureAwait(false);
+            }
+
+            // No remote host support, use inproc service
+            var service = document.GetLanguageService<ITodoCommentService>();
+            if (service == null)
+            {
+                // no inproc support
+                return SpecializedCollections.EmptyList<TodoComment>();
+            }
+
+            return await service.GetTodoCommentsAsync(document, tokens, cancellationToken).ConfigureAwait(false);
+        }
+
         private async Task<ImmutableArray<TodoItem>> CreateItemsAsync(Document document, IList<TodoComment> comments, CancellationToken cancellationToken)
         {
-            var items = ArrayBuilder<TodoItem>.GetInstance();
+            var items = ImmutableArray.CreateBuilder<TodoItem>();
```

Comment:
why not? it return immutable array and immutable array has builder so I made it to use it.